web/mediator/manifest.js

```diff
@@ -32,26 +37,37 @@ async function _getWebAppManifest({origin}) {
     }
     if(!cacheStorage) {
       // no cache storage/API available, fetch directly
-      return await _legacyFetch(url);
+      return await _legacyFetch(urls);
     }
 
-    // try to get cached response
```

Comment:
- `Promise.any()` may be a better choice than `race()`.
- Also consider passing in a `signal` and `AbortController`, and when one is done via `any`, cancel the rest.  I think the slower fetch might still continue if the result were never used?
- Does this need to fallback to load and parse html index and check for a manifest link and load that?  MDN even has an [example](https://developer.mozilla.org/en-US/docs/Web/Manifest#deploying_a_manifest) using `app.webmanifest`.

Reply:
@davidlehn,

> Also consider passing in a `signal` and `AbortController`, and when one is done via `any`, cancel the rest. I think the slower fetch might still continue if the result were never used?

We already have a 1 second timeout, so I don't think we need to do this -- and it would establish a stronger preference for whichever returned first vs. a preference by name, the latter might be easier for devs to reason about, but maybe it doesn't matter much since I doubt there's a use case for people to have those files out of sync to begin with.

> Does this need to fallback to load and parse html index and check for a manifest link and load that? MDN even has an [example](https://developer.mozilla.org/en-US/docs/Web/Manifest#deploying_a_manifest) using `app.webmanifest`.

No, I don't think so because we can't load the root HTML in most cases because of CORS restrictions, and we don't want to ask people to open up a potential vulnerability there.

Reply:
@dlongley 
> can we instead...

Cool, sounds good, that's coming in the next commit.

> could do something with Promise.race()

I'll probably avoid this because it returns the first completed promise, regardless if there is data or not. I'm still thinking `allSettled` might be best. While it does wait all promises to complete, but it returns their responses in the original order. Our 'prioritizing' can then be the order of the array of urls. The first one that comes back and contains data we'll use. If we put manifest.json at the 0th index of the urls array, that'll be first priority.